lib/bios/seabios.robot

```diff
@@ -0,0 +1,305 @@
+        ${line}=    Remove String Using Regexp    ${line}    ^[\\|\\s/\\\\-]+$
+        # If the resulting line is not empty, add it as a menu entry
+        ${length}=    Get Length    ${line}
+        IF    ${length} > 0    Append To List    ${construction}    ${line}
+    END
+    Log    ${construction}
+    ${construction}=    Get Slice From List    ${construction}    ${slice_start}    ${slice_end}
+    # TODO: Improve parsing of the menu into construction. It can probably be
+    # simplified, but at least we have this only in one kewyrod not in multiple
+    # ones.
+    # Make sure to remove control help text appearing in the screen if somehow
+    # they are still there.
+    Remove Values From List
+    ...    ${construction}
+    ...    Esc\=Exit
+    ...    ^v\=Move High
+    ...    <Enter>\=Select Entry
+    ...    F9\=Reset to Defaults F10\=Save
+    ...    LCtrl+LAlt+F12\=Save screenshot
+    ...    <Spacebar>Toggle Checkbox
+    ...    one adjusts to change
+    ...    Select boot device:
+    ...    , N for PXE boot
+    RETURN    ${construction}
+
+Enter Setup Menu SeaBIOS And Return Construction
+    [Documentation]    Enters Setup Menu and returns Setup Menu construction
+    Enter Setup Menu SeaBIOS
+    ${menu}=    Get Setup Menu Construction
+    RETURN    ${menu}
+
+Get RTC Clock Submenu Construction
+    [Arguments]    ${checkpoint}=Esc=Exit    ${lines_top}=1    ${lines_bot}=1    ${opt_only}="${FALSE}"
+    # In most cases, we need to strip two lines:
+    #    TOP:
+    #    Title line, such as:    Dasharo System Features
+    #    BOTTOM:
+    #    Help line, such as:    F9=Reset to Defaults    Esc=Exit
+    ${submenu}=    Get Menu Construction    ${checkpoint}    ${lines_top}    ${lines_bot}
+    # Handling of additional exceptions appearing in submenus:
+    #    1. Drop unselectable strings from Device Manager
+    Remove Values From List    ${submenu}    Devices List
+
+    IF    ${opt_only} == ${TRUE}
+        # Handling exceptions caused by some options splitting into multiple lines.
+        # For Dasharo System Features options, we can assume that each entry has
+        # either ">", or "[ ]", or "< >". For other edk2 menus, this is not always
+        # the case (yet?).
+        FOR    ${entry}    IN    @{submenu}
+            ${status}=    Check If Menu Line Is An Option    ${entry}
+            IF    ${status} != ${TRUE}
+                Remove Values From List    ${submenu}    ${entry}
+            END
+        END
+    END
+    RETURN    ${submenu}
+
+Get Option State
+    [Documentation]    Gets menu construction and option name as arguments.
+    ...    Returns option state, which can be: True or False.
+    [Arguments]    ${menu}    ${option}
+    ${index}=    Get Index Of Matching Option In Menu    ${menu}    ${option}
+    ${value}=    Get Regexp Matches    ${menu}[${index}]    (Enabled|Disabled)
+
+    RETURN    ${value}[0]
+
+Set Option State And Return Construction
```

Comment:
This is also not used in self-tests, or in any tests so far. Has it been tested? I am confused which keywords are really needed for seabios operation. Please confirm this.